python_ta/checkers/shadowing_in_comp_checker.py

```diff
@@ -0,0 +1,35 @@
+"""checker for variable shadowing in a comprehension.
+"""
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+
+class ShadowingInCompChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+
+    name = 'shadowing_inside_comprehension'
+    msgs = {'E9988': ("Comprehension loop variable '%s' shadows a variable in an outer scope",
+                      'shadowing-inside-comprehension',
+                      'Used when there is shadowing inside a comprehension'),
+            }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @check_messages('shadowing-inside-comprehension')
+    def visit_comprehension(self, node):
+        if isinstance(node.target, astroid.Tuple):
+            for targ in node.target.elts:
+                if targ.name in node.parent.frame().locals:
+                    args = targ.name
+                    self.add_message('shadowing-inside-comprehension', node=targ, args=args)
+        # else: isinstance(node.target, astroid.AssignName)
```

Comment:
You really need the else here, otherwise the block below will always run, even when `node.target` isn't an AssignName.